src/platform/app_annotations.nit

```diff
@@ -0,0 +1,132 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Annotations to gather metadata on `app.nit` projects
+module app_annotations
+
+private import parser_util
+import modelize
+import literal
+import semantize
+private import annotation
+
+# Metadata associated to an `app.nit` project
+class AppProject
+	# Pretty name of the resulting application
+	var name: String = mainmodule.first_real_mmodule.name is lazy
+
+	# Short project used in `namespace` and configuration files
+	var short_name: String = mainmodule.name.replace("-", "_") is lazy
+
+	# Namespace/package used to identify the application
+	var namespace = "org.nitlanguage.{short_name}" is lazy
+
+	# Version of the application
+	var version = "0.1"
+
+	# Numerical version code of th application
+	var version_code: Int is lazy do
+
```

Comment:
Me too! but sadly the grammar does not support moving the `do` on the next line.